content/docs/development/events.md

```diff
@@ -0,0 +1,40 @@
+# Events
+
+Cartographer's controller emits
+[Kubernetes Events](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#event-v1-core) when reconciling
+owner resources (those belonging to a Workload, Deliverable or Runnable) objects.
+
+## Events in owner resources
+
+Cartographer emits events with regard to specific resources belonging to a Workoad, Deliverable or Runnable.
+
+Events may be emitted for the following reasons:
+
+- `StampedObjectApplied` this event is emitted whenever Cartographer creates or updates an owner resource.
+- `StampedObjectRemoved` this event is emitted whenever Cartographer removes a resource.
+- `ResourceOutputChanged` this event is emitted whenever Cartographer notices a resource's output has changed since the
+  last time it was observed.
+- `ResourceHealthyStatusChanged` this event is emitted whenever Cartographer notices the healthy status of a resources
+  has changed. For more details see [Health Rules](health-rules.md)
+
+### Example events
+
+Here's an example of events retrieved from a workoad using `kubectl describe workload <my-workload> -n <ns>`:
+
+```
+... Workload description redacted ...
```

Comment:
I just ran an example here with the following Runnable

```yaml
---
apiVersion: carto.run/v1alpha1
kind: Runnable
metadata:
  name: test
spec:
  retentionPolicy:
    maxFailedRuns: 1
    maxSuccessfulRuns: 1

  runTemplateRef:
    name: tekton-taskrun

  inputs:
    blob-url: https://github.com/carto-run/hello-world
    blob-revision: 19769456b6b229b3e78f2b90eced15a353eb4e7c
```

(same as https://github.com/vmware-tanzu/cartographer/tree/main/examples/runnable-tekton but adding retention policy) 

and thad the following set of events after tweaking an input here and there to force gc to also happen:

```
Events:
  Type    Reason                 Age    From      Message
  ----    ------                 ----   ----      -------
  Normal  StampedObjectApplied   4m5s   Runnable  Created object [taskruns.tekton.dev/test-sg4m5]
  Normal  StampedObjectApplied   3m49s  Runnable  Created object [taskruns.tekton.dev/test-mx7nw]
  Normal  ResourceOutputChanged  3m37s  Runnable  Runnable [test] found a new output in [taskruns.tekton.dev/test-mx7nw]
  Normal  StampedObjectApplied   119s   Runnable  Created object [taskruns.tekton.dev/test-hzkq2]
  Normal  StampedObjectRemoved   113s   Runnable  Deleted object [taskruns.tekton.dev/test-mx7nw]
  Normal  ResourceOutputChanged  113s   Runnable  Runnable [test] found a new output in [taskruns.tekton.dev/test-hzkq2]
```

wdyt of having that as a secondary example? i.e., one that covers Runnable (aside from these ones here that cover Workload)?

Reply:
Yeah, good call out. This way we have all the documented event resasons represented.